proof/file.go

```diff
@@ -32,6 +36,27 @@ type Version uint32
 const (
 	// V0 is the first version of the proof file.
 	V0 Version = 0
+
+	// FileMaxNumProofs is the maximum number of proofs we expect/allow to
+	// be encoded within a single proof file. Given that there can only be
+	// one transfer per block, this value would be enough to transfer an
+	// asset every 10 minutes for 1 year straight. This limitation might be
+	// lifted at some point when proofs can be compressed into a single
+	// zero-knowledge proof.
+	FileMaxNumProofs = 52500
```

Comment:
I think maybe we'll want a bit more breathing room here. 

Mid-tern, we can also move to a route where we just keep them on disk and memory map, so then we never have to load the entire thing into user space memory at once.

Reply:
Yeah, makes sense. Bumped it to 420000 (roughly 8 years).

Reply:
I imagine we won't write proof files for each update in an LN channel right?

Reply:
No, I don't think so. We'll just create a new single transition proof suffix for each update, and each update would be at the same "height" (same location in the proof file, so the file wouldn't be extended with every update).

Reply:
> I imagine we won't write proof files for each update in an LN channel right?

They'll get replaced basically each time. Both sides also have everything they need to make the proof files that result from on-chain channel operations.

Reply:
Hmm so if rewrite + update DB num proofs would be static 👌🏽